rewrite-java/src/main/java/org/openrewrite/java/RenamePackage.java

```diff
@@ -0,0 +1,320 @@
+                                || (!(((J.FieldAccess) getCursor().getParent().getValue())
+                                        .isFullyQualifiedClassReference(targetPackageType.getFullyQualifiedName()))))) {
+
+                    f = TypeTree.build(((JavaType.FullyQualified) targetPackageType).getFullyQualifiedName())
+                            .withPrefix(f.getPrefix());
+                }
+            }
+            return f;
+        }
+
+        @Override
+        public J.Identifier visitIdentifier(J.Identifier ident, ExecutionContext ctx) {
+            // If the J.Identifier's type is equal to the type we're looking for,
+            // and the classname of the type we're looking for is equal to the J.Identifiers string representation
+            // Then transform it, otherwise leave it alone
+            J.Identifier i = super.visitIdentifier(ident, ctx);
+
+            if (TypeUtils.isOfClassType(i.getType(), oldPackageName)
+                    && i.getSimpleName().equals(originalPackageType.getClassName())) {
+
+                i = i.withName((targetPackageType).getClassName())
+                        .withType(targetPackageType);
+            }
+            return i;
+        }
+
+        @Override
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+            J.MethodDeclaration m = super.visitMethodDeclaration(method, ctx);
+            m = m.withReturnTypeExpression(transformName(m.getReturnTypeExpression()));
+            return m.withThrows(m.getThrows() == null ? null : ListUtils.map(m.getThrows(), this::transformName));
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+            J.MethodInvocation m = super.visitMethodInvocation(method, ctx);
+
+            if (m.getSelect() instanceof NameTree
+                    && m.getType() != null && m.getType().hasFlags(Flag.Static)) {
+                m = m.withSelect(transformName(m.getSelect()));
+            }
+
+            if (m.getSelect() != null) {
+                JavaType.FullyQualified selectType = TypeUtils.asFullyQualified(m.getSelect().getType());
+                if (selectType != null && selectType.getFullyQualifiedName().equals(oldPackageName)) {
+                    m = m.withSelect(m.getSelect()
+                            .withType(targetPackageType));
+                }
+            }
+
+            if (m.getType() != null) {
+                if (m.getType().getDeclaringType().getFullyQualifiedName().equals(oldPackageName)) {
+                    m = m.withDeclaringType(targetPackageType);
+                }
+            }
+
+            return m;
+        }
+
+        @Override
+        public J.MultiCatch visitMultiCatch(J.MultiCatch multiCatch, ExecutionContext ctx) {
+            final J.MultiCatch m = visitAndCast(multiCatch, ctx, super::visitMultiCatch);
```

Comment:
just curious why not just call `super.visitMultiCatch`?

Reply:
I missed this in the multicatch and wildcard - will make the changes, thanks!